lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -124,6 +122,38 @@ SdfLayerHandle defPrimSpecLayer(const UsdPrim& prim)
 	return defLayer;
 }
 
+bool isTargetLayerHaveOpinion(const UsdPrim& prim)
+{
+    auto editTarget = prim.GetStage()->GetEditTarget();
+    auto layer = editTarget.GetLayer();
+    auto primSpec = layer->GetPrimAtPath(prim.GetPath());
+
+    // to know whether the target layer contains any opinions that 
+    // affect a particular prim, there must be a primSpec for that prim
+    if (!primSpec) {
+        return false;
+    }
+
+    return true;
+}
+
+SdfLayerHandle targetLayerWithOpion(const UsdPrim& prim)
+{
+    SdfLayerHandle targetLayer;
+    auto layerStack = prim.GetStage()->GetLayerStack();
+    for (auto layer : layerStack)
+    {
+        // to know whether the target layer contains any opinions that 
+        // affect a particular prim, there must be a primSpec for that prim
+        auto primSpec = layer->GetPrimAtPath(prim.GetPath());
+        if (primSpec) {
+            targetLayer = layer;
+            break;
+        }
+    }
+    return targetLayer;
+}
+
```

Comment:
This function is used to give the user a hint to what target layer to set in order to rename a particular prim.